R/dataProcessPlots.R

```diff
@@ -491,4 +602,140 @@ dataProcessPlots = function(
+        if(is_bool) plot$x$data[[i]]$showlegend <- FALSE
+    }
+    plot
+
+}
+
+.fix.censored.points.legend.profileplots.plotly = function(plot) {
+    df <- data.frame(id = seq_along(plot$x$data), legend_entries = unlist(lapply(plot$x$data, `[[`, "name")))
+    detected_data = FALSE
+    censored_data = FALSE
+    for (i in df$id) {
+        if(!detected_data && df$legend_entries[[i]] == "FALSE") {
+            plot$x$data[[i]]$name <- "Detected data"
+            plot$x$data[[i]]$showlegend <- TRUE
+            # plot$x$data[[i]]$marker$color <- 'rgba(0,0,0,1)'
+            # plot$x$data[[i]]$marker$line$color <- 'rgba(0,0,0,1)'
+            detected_data = TRUE
+
+        }
+        if(!censored_data && df$legend_entries[[i]] == "TRUE") {
+            plot$x$data[[i]]$name <- "Censored missing data"
+            plot$x$data[[i]]$showlegend <- TRUE
+            # plot$x$data[[i]]$marker$color <- 'rgba(0,0,0,1)'
+            # plot$x$data[[i]]$marker$line$color <- 'rgba(0,0,0,1)'
+            censored_data = TRUE
+        }
+    }
+    plot
+}
+
+.fix.legend.plotly.plots.volcano = function(plot) {
+    df <- data.frame(id = seq_along(plot$x$data), legend_entries = unlist(lapply(plot$x$data, `[[`, "name")))
+    # Create a mapping
+    color_mapping <- c("black" = "No regulation", "red" = "Up-regulated", "blue" = "Down-regulated")
+    # Update the legend_entries column
+    df$legend_group <- sapply(df$legend_entries, function(entry) {
+        for (color in names(color_mapping)) {
+            if (grepl(color, entry)) {
+                entry <- gsub(color, color_mapping[color], entry)
+                break
+            }
+        }
+        entry <- gsub(",.+", "", entry)
+        entry <- gsub("\\(|\\)", "", entry)  # Remove any remaining parentheses
+        entry
+    })
+    for (i in df$id) {
+        if(length(grep(df$legend_group[[i]], color_mapping)) == 0) { # keep only 3 legends
+            plot$x$data[[i]]$showlegend <- FALSE
+        }
+        plot$x$data[[i]]$name <- df$legend_group[[i]]
+        plot$x$data[[i]]$legendgroup <- plot$x$data[[i]]$name
+    }
+    plot <- plot %>% plotly::layout(legend=list(title=list(text="")))
+    plot
+}
+
+#' @export
+.get.plotly.plot.html = function(plots, width, height) {
+    doc <- htmltools::tagList(lapply(plots,function(x) htmltools::div(x, style = "float:left;width:100%;")))
+    # Set a specific width for each plot
+    plot_width <- 800
+    plot_height <- 600
+
+    # Create a div for each plot with style settings
+    divs <- lapply(plots, function(x) {
+        # htmltools::div(x, style = paste0("width:", plot_width, "; display:inline-block;"))
+        htmltools::div(x, style = paste0("width:", plot_width, "px; height:", plot_height, "px; margin: 10px;"))
+    })
+
+    # Combine the divs into a tagList
+    doc <- htmltools::tagList(divs)
+    doc
+}
+
+#' @export
+.save.plotly.plot.html = function(plots, address, file_name, width, height) {
```

Comment:
naming

Reply:
Commit [fix name and exports](https://github.com/Vitek-Lab/MSstats/pull/107/commits/5d88fc0353c4a1e3ae1add4b068eec4a91ef93da)